### PR TITLE
Gce/es24

### PR DESCRIPTION
**Pull request by lukasz-golebiewski**
This PR is only for review purposes, it is not going to be merged. The code is going to be left on the `gce/es24` branch.

### CODE REVIEW

**Comment by lukasz-golebiewski** [comment]
Pushed as `gwidx/elasticsearch:2.4.5-gce`

File: config/elasticsearch.yml [hunk 1]
```diff
@@ -1,46 +1,51 @@
...
 discovery:
-  zen.ping.multicast.enabled: false
-  zen.minimum_master_nodes: ${ES_MINIMUM_MASTER_NODES}
-  type: ec2
-  ec2:
-    tag:
-      Cluster: ${ES_CLUSTER_NAME}
+  type: kubernetes
+  zen:
+    ping.unicast.hosts: ${DISCOVERY_SERVICE}
```
Comment (lustefaniak) [inline config/elasticsearch.yml]: is this needed?

File: Dockerfile [hunk 2]
```diff
@@ -1,37 +1,48 @@
...
+CACHE_FIELD_DATA_SIZE=30% \
+ES_JAVA_OPTS="-Xms512m -Xmx512m"
+
+# Set environment variables defaults
+ENV NETWORK_HOST _site_
+ENV HTTP_CORS_ENABLE true
+ENV HTTP_CORS_ALLOW_ORIGIN *
+ENV NUMBER_OF_MASTERS 1
+ENV MAX_LOCAL_STORAGE_NODES 1
+ENV SHARD_ALLOCATION_AWARENESS ""
```
Comment (lustefaniak) [inline Dockerfile]: Those are not used anywhere

File: Dockerfile [hunk 3]
```diff
@@ -1,37 +1,48 @@
...
+STORE_THROTTLE_MBPS=20mb \
+CACHE_FIELD_DATA_SIZE=30% \
+ES_JAVA_OPTS="-Xms512m -Xmx512m"
+
+# Set environment variables defaults
+ENV NETWORK_HOST _site_
+ENV HTTP_CORS_ENABLE true
+ENV HTTP_CORS_ALLOW_ORIGIN *
+ENV NUMBER_OF_MASTERS 1
+ENV MAX_LOCAL_STORAGE_NODES 1
```
Comment (lustefaniak) [inline Dockerfile]: What is that?

File: Dockerfile [hunk 4]
```diff
@@ -1,37 +1,48 @@
...
-ES_WARMER_ENABLED=true \
-ES_STORE_THROTTLE_MBPS=20mb \
-ES_CACHE_FIELD_DATA_SIZE=30%
+INDEX_BUFFER_SIZE=10% \
+CACHE_FILTER_SIZE=10% \
+BREAKER_TOTAL_LIMIT=70% \
+BREAKER_FIELD_DATA_LIMIT=60% \
+BREAKER_REQUEST_LIMIT=40% \
+WARMER_ENABLED=true \
+STORE_THROTTLE_MBPS=20mb \
```
Comment (lustefaniak) [inline Dockerfile]: Probably on GCE on SSD this could be changed to something higher

**Comment by lustefaniak** [comment]
I guess it could be worth to check what settings defines image which Petr deployed, maybe it is already optimized for GCE use and we should only install plugins but not alter other settings

**Comment by lukasz-golebiewski** [comment]
The elasticsearch repo from `pires` doesn't event have version `2.4.5`, the latest is `2.4.1` hence the need for adaptation. `2.4.1` doesn't contain any optimisations. Doing what you proposed would make sense only if we moved to ES version `5.x.y`

File: Dockerfile (hunk 3 above)
Comment (lukasz-golebiewski) [inline Dockerfile]: https://github.com/elastic/elasticsearch/issues/8932

File: Dockerfile (hunk 2 above)
Comment (lukasz-golebiewski) [inline Dockerfile]: True, removed

File: config/elasticsearch.yml (hunk 1 above)
Comment (lukasz-golebiewski) [inline config/elasticsearch.yml]: Apparently yes

**Pull request closed without merging**